src/api/challenges.ts

```diff
@@ -0,0 +1,74 @@
+// challenges.ts - Router for the challenges model
+
+// import dependencies
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+
+const challenges = express.Router();
+
+const prisma = new PrismaClient();
+
+// Might be unnecessary with current frontend requirements
+// Example middleware to filter out 404 errors early
+challenges.use("/:id/*", async (req, res, next) => {
+  try {
+    // Try to parse the id into a number, if it is NaN, return 400 error
+    const id = parseInt(req.params.id);
+    if (Number.isNaN(id)) {
+      res.status(400).json({ message: "Bad Request, ids must be integers" });
+      return;
+    }
+
+    // Once the id is parsed, retrieve the corresponding challenge from the database
+    // If no challenges have a matching id, return a 404 error and return to stop execution.
+
+    let challenge;
+    challenge = await prisma.challenge.findUnique({ where: { id: id } });
+
+    if (challenge === null) {
+      res
+        .status(404)
+        .json({ message: `Not Found, requested resource was not found` });
+      return;
+    }
+
+    // Add the requested challenge to the body of the request
+    req.body.challenge = challenge;
```

Comment:
Are we deleting challenges ever?

Reply:
Potentially. There are get, delete and update methods on SwaggerHub, but they are not MVP. Do you think we should just remove this?

Reply:
Probably if we're deleting challenges through the API in the future. We're implementing admin and user levels for route protection, right?